packages/blockchain-wallet-v4-frontend/src/data/modals/slice.ts

```diff
@@ -0,0 +1,47 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { merge } from 'ramda'
+
+import { ModalNameType, ModalsState, UpdateModalOptionsPayload, UpdateModalPayload } from './types'
+
+const initialState: ModalsState = []
+
+const modalsSlice = createSlice({
+  initialState,
+  name: 'modals',
+  reducers: {
+    closeAllModals: (state) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      state = []
+    },
+    closeModal: (state, action: PayloadAction<{ modalName?: ModalNameType }>) => {
+      if (action.payload.modalName) {
+        state = state.splice(
```

Comment:
Since the splice method just modifies an array in place you can remove the `state = ` and just modify the state array directly.